README.md

```diff
@@ -1,2 +1,133 @@
-# sparc-request-kubernetes
-Informational repostiory for building and running SPARCRequest for Kubernetes
+# SPARCRequest - Kubernetes
+
+This reference project can be used to inform how your group could run SPARCRequest in a Kubernetes environment. The following is representative how OCTRI chose to implement SPARCRequest.
+
+## Considerations for running SPARCRequest in Kubernetes
+
+### SPARCRequest Customization
+
+By using a second image we are able to modify our instance of SPARCRequest without needing to maintain a separate branch in source control.
+
+See the Image section below for a description how we build and manage our SPARCRequest instance.
+
+### Deployments
+
+We have two deployment workloads defined; one for the main application and a second for the delayed job that performs background tasks like sending emails.
+
+#### Main Deployment
+
+The main SPARCRequest application generally runs well inside Kubernetes, a single process Puma web server running the rails app. As seen in the [`deployment`](./k8s/deployment.yaml) manifest, we specify the resources, configuration, liveness, initContainers and any volumes required for persistence.
+
+The `initContainers` are used to perform database migrations and asset compilation.
+
+The volumes persist the things like attachments. We use a NFS  `PersistentVolume` to preserve these files beyond the lifespan of a pod.
+
+The `livenessProbe` informs the scheduler when the new pod is up and that it can terminate the old pod. This is important because the database migrations and asset compilation can take a bit of time and this reduces the time the application is unavailable.
```

Comment:
The primary purpose of the liveness probe is to ensure that the application is restarted if it becomes unresponsive. What you're describing here is a readiness probe.

I thought at some point that I confirmed that the liveness probe would be used to check readiness if no readiness probe was specified, but from [the documentation](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes) that doesn't appear to be the case:

> `readinessProbe`: Indicates whether the container is ready to respond to requests. If the readiness probe fails, the endpoints controller removes the Pod's IP address from the endpoints of all Services that match the Pod. The default state of readiness before the initial delay is `Failure`. If a Container does not provide a readiness probe, the default state is `Success`.

So to get the best behavior out of our deployments, we should actually define a readiness probe. In practice this hasn't really been an issue though, because the SPARC container starts in just a few seconds.